docs/word/word-add-ins-core-concepts.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Fundamental programming concepts with the Word JavaScript API
+description: Use the Word JavaScript API to build add-ins for Word.
+ms.date: 07/05/2019
+localization_priority: Priority
+---
+
+# Fundamental programming concepts with the Word JavaScript API
+
+This article describes concepts that are fundamental to using the [Word JavaScript API](../reference/overview/word-add-ins-reference-overview.md) to build add-ins for Word 2016 or later.
+
+## Asynchronous nature of Word APIs
+
+The Word JavaScript API is loaded by Office.js. The Word JavaScript API changes the way that you can interact with objects like documents and paragraphs. Rather than providing individual asynchronous APIs for retrieving and updating each of these objects, the Word JavaScript API provides "proxy" JavaScript objects that correspond to the real objects running in Word. You can interact with these proxy objects by synchronously reading and writing their properties and calling synchronous methods to perform operations on them. These interactions with proxy objects aren't immediately realized in the running script. The **context.sync** method synchronizes the state between your running JavaScript and the real objects in Office by executing queued instructions and retrieving properties of loaded Word objects for use in your script.
```

Comment:
Should "context.sync" be in code ticks instead of bolded? There's a recurring theme of that in this document, so check for consistency in you make any changes.

Reply:
I'll create a separate work item for ensuring that all code element strings are formatted as code (not bold); this issue is present in a bunch of places throughout the docs and it'll be too time to consuming to address now as part of this PR.

Reply:
I think there may already be such an item...